pkg/lib/cockpit-components-select.jsx

```diff
@@ -20,125 +20,184 @@
-            if (this.props.enabled === false)
+            }
+            let buttonClasses = "btn btn-default dropdown-toggle";
+            if (this.props.enabled === false) {
                 buttonClasses += " disabled";
+            }
 
             return (
-                <div className={classes} onClick={this.clickHandler} id={this.props.id}>
+                <div className={classes} onClick={this.clickHandler.bind(this)} id={this.props.id}>
                     <button className={buttonClasses} type="button">
                         <span className="pull-left">{currentValue}</span>
                         <span className="caret"></span>
                     </button>
                     <ul className="dropdown-menu">
-                        { listItems }
+                        {listItems}
                     </ul>
                 </div>
             );
         }
-    });
+    }
+
+    StatelessSelect.propTypes = {
+        selected: React.PropTypes.string,
+        onChange: React.PropTypes.func,
+        id: React.PropTypes.string,
+        enabled: React.PropTypes.bool,
+    };
+
+    class Select extends React.Component {
+        constructor(props) {
+            super();
+
+            this.state = {
+                currentData: props.initial,
+            };
+        }
+
+        onChange(data) {
+            this.setState({ currentData: data });
+            typeof this.props.onChange === 'function' && this.props.onChange(data);
+        }
+
+        render() {
+            return (
+                <StatelessSelect onChange={this.onChange.bind(this)}
+                                 selected={this.state.currentData}
+                                 id={this.props.id}
+                                 enabled={this.props.enabled}>
+                    {this.props.children}
+                </StatelessSelect>
+            )
+        }
+
+    }
+
+    Select.propTypes = {
+        initial: React.PropTypes.string,
+        onChange: React.PropTypes.func,
+        id: React.PropTypes.string,
+        enabled: React.PropTypes.bool,
+    };
 
     /* Entry class for the select component
      * Dynamic lists should make sure to also provide 'key' props for react to use
      * Expected properties:
-     *  - data optional, will be passed to the select's onChange callback
+     *  - data (required), will be passed to the select's onChange callback
      * Example: <SelectEntry data="foo">Some entry</SelectEntry>
      */
-    var SelectEntry = React.createClass({
-        propTypes: {
-            data: React.PropTypes.string.isRequired,
-        },
-        render: function() {
-            var value = (this.props.children !== undefined) ? this.props.children : textForUndefined;
+    class SelectEntry extends React.Component {
+        render() {
+            const value = (this.props.children !== undefined) ? this.props.children : textForUndefined;
             return <a>{value}</a>;
```

Comment:
This is a presentational component and thus could just be a function. (Doesn't matter much, though).

More importantly, it seems the required "data" property is not used at all? This seems to do nothing else than just show its children.